<?xml version="1.0"?>

<launch>

    <!-- Add intensity to pointcloud based on image within simulation -->
    <node pkg="line_detection_vlp16" type="fake_intensity_pcd.py" name="fake_intensity_pcd" output="screen">
        <remap from="/points_in"    to="/velodyne_points"/>
        <remap from="/points_out"   to="/velodyne_points_sim"/>

        <param name="map_image_path"    value="$(find igvc2023)/models/course_v2/materials/textures/igvc2022_course_image_v2_1400x1200.png"/>
        <param name="map_size"          value="43 37"/>
        <param name="origin"            value="0 16.5"/>
        <param name="urdf_model_name"   value="orange"/>
        <param name="sensor_height"     value="0.8015"/>
    </node>

    <!-- Publish detected line as pointcloud -->
    <node pkg="line_detection_vlp16" type="line_detection.py" name="line_detection" output="screen">
        <remap from="/points_in"    to="/velodyne_points_sim"/>

        <rosparam command="load" file="$(find line_detection_vlp16)/param/line_detection_params.yaml" />
    </node>

    <!-- <node pkg="line_detection_vlp16" type="plot_cost.py" name="plot_cost" output="screen" /> -->

    <!-- Dynamic reconfigure -->
    <node pkg="rqt_gui" type="rqt_gui" name="reconfigure" args="-s reconfigure"/>

</launch>